### PR TITLE
Add support for python 3.6

### PR DESCRIPTION
**Pull request by chrisseto**

### CODE REVIEW

File: invoke/tasks.py [hunk 1]
```diff
@@ -132,10 +132,20 @@ def argspec(self, body):
...
+        except AttributeError:
+            signature = inspect.signature(func)
+            arg_names = list(signature.parameters.keys())
+            spec_dict = {
+                key: value.default or NO_DEFAULT for
+                key, value in signature.parameters.items()
+            }
+        else:
+            arg_names = spec.args[:]
+            matched_args = [reversed(x) for x in [spec.args, spec.defaults or []]]
```
Comment (betterdiff) [inline invoke/tasks.py]: * E501 line too long (82 > 79 characters)

**Comment by nicktimko** [comment]
Hah, it seems like Python 3.6 isn't Python 3.6 compatible as-of-yet, committing the same error you're fixing:
```
...
File "/home/travis/virtualenv/python3.6-dev/lib/python3.6/site-packages/pep8.py", line 1186, in register_check
    args = inspect.getargspec(check)[0]
AttributeError: module 'inspect' has no attribute 'getargspec'
```

**Comment by nicktimko** [comment]
This will require an upgraded flake8; it crashes because pep8 crashes. pep8 master (1.6.3a0) is up-to-date with improved inspect (Python 3.3+ with signature objects) support, but [flake8 2.4.0 restricts it to <1.6](https://gitlab.com/pycqa/flake8/blob/master/CHANGES.rst), likewise with 2.5.0 (current release on PyPI)...though actually it just says [*not* pep8 1.6.0, .1, or .2](https://gitlab.com/pycqa/flake8/blob/master/setup.py#L53).

**Comment by rpkilby** [comment]
This looks like it can be closed - py36 is now included in the CI matrix.

**Pull request closed without merging**